datafeed-dapp/contracts/Dex.sol

```diff
@@ -1,147 +1,83 @@
 
 contract Dex {
     address public immutable owner;
-    IDelegator delegator;
-
-    address public token1;
-    address public token2;
-
-    uint16 public token1ID;
-    uint16 public token2ID;
-
-    uint256 fundAmount = 0.01 ether;
-
-    event Swap(
-        uint16 fromID,
-        uint16 toID,
-        uint256 fromAmount,
-        uint256 toAmount
-    );
+    ITransparentForwarder public transparentForwarder;
+    address public usdToken;
+    bytes32 public collectionNameHash;
 
     modifier onlyOwner() {
         require(msg.sender == owner, "caller is not the owner");
         _;
     }
 
-    modifier checkID(uint16 _id) {
-        require(_id == 1 || _id == 2, "only id 1 and 2 can be updated");
-        _;
-    }
-
     constructor(
-        address _delegator,
-        address _token1,
-        address _token2,
-        uint16 _token1ID,
-        uint16 _token2ID
+        address _transparentForwarder,
+        address _usdToken,
+        bytes32 _collectionNameHash
     ) {
         owner = msg.sender;
-        delegator = IDelegator(_delegator);
-
-        token1 = _token1;
-        token2 = _token2;
 
-        token1ID = _token1ID;
-        token2ID = _token2ID;
+        transparentForwarder = ITransparentForwarder(_transparentForwarder);
+        usdToken = _usdToken;
+        collectionNameHash = _collectionNameHash;
     }
 
-    /// @notice Swap ERC20 tokens
-    /// @dev Swap ERC20 tokens based on oracle result
-    /// @param _from From tokenID
-    /// @param _to To tokenID
-    /// @param _amount Amount of from tokens
-    function swap(
-        uint16 _from,
-        uint16 _to,
-        uint256 _amount
-    ) public {
-        require(
-            (_from == token1ID && _to == token2ID) ||
-                (_from == token2ID || _to == token1ID),
-            "invalid token pair"
-        );
-
-        address from = token1ID == _from ? token1 : token2;
-        address to = token2ID == _to ? token2 : token1;
-
-        uint256 toAmount = getSwapAmount(_from, _to, _amount);
-        emit Swap(_from, _to, _amount, toAmount);
-
-        IERC20(from).transferFrom(msg.sender, address(this), _amount);
-        IERC20(to).approve(address(this), toAmount);
-        IERC20(to).transferFrom(address(this), msg.sender, toAmount);
-    }
+    /// @notice Swap native token with equivalent USD token
+    function swap() public payable {
+        uint256 collectionResult;
+        int8 collectionPower;
+        uint256 collectionPriceInUSD;
+
+        (collectionResult, collectionPower) = transparentForwarder.getResult{
+            value: 0
+        }(collectionNameHash);
+
+        // * if power is +ve, price = result / 1o^power
+        // * if power is -ve, price = result * 10^power
+        if (collectionPower < 0) {
+            collectionPriceInUSD =
+                collectionResult *
+                uint256(pow(collectionPower));
+        } else {
+            collectionPriceInUSD =
```

Comment:
multiply with msg.value with result and then divide

Reply:
Resolved in [c9f3c35](https://github.com/razor-network/razor-network-examples/pull/5/commits/c9f3c3534e149c4ee881864eededcc5d40e400c7)